prototyping/src/compiler.py

```diff
@@ -0,0 +1,23 @@
+from pathlib import Path
+from subprocess import PIPE, STDOUT, run
+
+SRC = Path(__file__).resolve().parent
+
+
+class CompilationError(Exception):
+    pass
+
+
+def compile_lib(source: Path, cflags=[], ldadd=[]):
+    binary = source.with_suffix(".so")
+    result = run(
+        ["gcc", "-shared", *cflags, "-o", str(binary), str(source), *ldadd],
```

Comment:
We may have to use the `-I` option here to provide the path to the header file of QVector, as explained [in this comment](https://stackoverflow.com/a/17475328). I tried it though, and couldn't make it work.

Reply:
At the moment where we are willing to have a small library that contains/uses QVector, we need to compile it with Qt. Probably using qmake here is easier